src/sql/src/query_model/hir.rs

```diff
@@ -274,6 +274,19 @@ impl FromHir {
                 then: Box::new(self.generate_expr(*then, context_box)),
                 els: Box::new(self.generate_expr(*els, context_box)),
             },
+            HirScalarExpr::Select(expr) => {
+                let box_id =
+                    self.within_context(context_box, expr, &mut |generator, expr| -> BoxId {
```

Comment:
Instead of adding an extra argument to within_context, is it possible to capture expr by value using [move](https://doc.rust-lang.org/std/keyword.move.html)?

Reply:
Done. I tried that, but ran into the issue of not being able to take `expr` out of the `Box` it lives in. It turns out it was as simple as calling `expr.take()`